### PR TITLE
Add mkl dump functionality 

### PR DESCRIPTION
**Pull request by leila-pujal**
I added `dump_one` function for molekel file format and tested it similar to molden `dump_one` function.

### CODE REVIEW

**Comment by tovrstra** [comment]
This looks already pretty good. Could you rename the functions `check_load_molden_dump_molekel_li2` and `check_load_fchk_dump_molekel_li2` to `test_load_molden_dump_molekel_li2` and `test_load_fchk_dump_molekel_li2`? If they are not meant to pass yet, you can always mark them with xfail: https://docs.pytest.org/en/latest/skipping.html#xfail-mark-test-functions-as-expected-to-fail Thanks.

**Comment by leila-pujal** [comment]
`Check_load_molden_dump_molekel_li2` and `check_load_fchk_dump_molekel_li2` were changed to `test_load_molden_dump_molekel_li2` and `test_load_fchk_dump_molekel_li2`. In order to have these tests to pass, a new `compare_mols` function is defined in test_molekel.py, `compare_mols_diff_formats`.  It was not possible to use `compare_mols` function in common.py, as it is supposed to compare two IOData objects created from the same type of format file.  Also, some minor changes were made in molekel.py, related to atomic basis information for each center and saving/printing of irreps for orbital symmetry.

File: iodata/formats/molekel.py [hunk 1]
```diff
@@ -112,7 +126,8 @@ def _load_helper_coeffs(lit: LineIterator, nbasis: int) -> Tuple[np.ndarray, np.
             ncol = len(words)
             assert ncol > 0
             for word in words:
-                assert word == 'a1g'
+                # assert word == 'a1g'
```
Comment (tovrstra) [inline iodata/formats/molekel.py:129]: Feel free to remove this commented line. This assert should never have been here in the first place.

File: iodata/formats/molekel.py [hunk 2]
```diff
@@ -223,13 +243,150 @@ def load_one(lit: LineIterator) -> dict:
...
+        f.write('\n')
+
+        # COEFF_BETA
+        f.write('$COEFF_BETA\n')
+        _dump_helper_coeffs(f, data, spin='b')
+
+        # OCC_BETA
+        f.write('$OCC_BETA\n')
+        _dump_helper_occ(f, data, spin='b')
+
```
Comment (tovrstra) [inline iodata/formats/molekel.py:339]: If you want to deal with the generalized case, you might add something like

```python
elif data.mo.kind == 'generalized':
    raise ValueError("The MKL format does not support generalized orbitals..")
```

Comment (FarnazH) [reply iodata/formats/molekel.py:338]: Thanks for your comment. Fixed it with an `else` statement to be more general.

File: iodata/formats/molekel.py (hunk 1 above)
Comment (FarnazH) [reply iodata/formats/molekel.py]: Removed. Thanks.

**Pull request merged**